### PR TITLE
feat(tier4_system_msgs): add mode change available message 

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Related Links

https://github.com/autowarefoundation/autoware.universe/pull/2120

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Code is properly formatted
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code is properly formatted
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR**: Required to pass before the merge.
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: tier4_system_msgs/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,7 @@ rosidl_generate_interfaces(${PROJECT_NAME}
   "msg/HazardStatus.msg"
   "msg/HazardStatusStamped.msg"
   "msg/ServiceLog.msg"
+  "msg/ModeChangeAvailable.msg"
```
Comment (kenji-miyake) [inline tier4_system_msgs/CMakeLists.txt:24]: Considering there is already `IsAutonomousAvailable`, is adding `Is` better?

```suggestion
  "msg/IsModeChangeAvailable.msg"
```

File: tier4_system_msgs/msg/ModeChangeAvailable.msg [hunk 2]
```diff
@@ -0,0 +1,2 @@
+builtin_interfaces/Time stamp
+bool available
```
Comment (kenji-miyake) [inline tier4_system_msgs/msg/ModeChangeAvailable.msg:2]: To be consistent with https://github.com/tier4/tier4_autoware_msgs/blob/414af45fb1e786e81a269ded967c2edc871a32cd/tier4_system_msgs/msg/IsAutonomousAvailable.msg#L2.

```suggestion
bool is_available
```

File: tier4_system_msgs/CMakeLists.txt (hunk 1 above)
Comment (isamu-takagi) [reply tier4_system_msgs/CMakeLists.txt:24]: Use `ModeChangeAvailable` as a generalization of `IsAutonomousAvailable` (for remote and local modes). I'll remove `IsModeChangeAvailable` later.

File: tier4_system_msgs/msg/ModeChangeAvailable.msg (hunk 2 above)
Comment (isamu-takagi) [reply tier4_system_msgs/msg/ModeChangeAvailable.msg:2]: I'll remove `IsModeChangeAvailable` later.

**Review by kenji-miyake — APPROVED** [review]
It might be better to use `ModeChangeAvailability` and the field `data`, but since there is no guideline for them now, I'll leave it to you for the namings.

**Comment by isamu-takagi** [comment]
I would like to consider the name along with the guidelines.

**Pull request merged**